### PR TITLE
feat: use github.com/mdlayher/kobject

### PR DESCRIPTION
**Pull request by andrewrynhard**
Signed-off-by: Andrew Rynhard <andrew@andrewrynhard.com>

### CODE REVIEW

**Comment by andrewrynhard** [comment]
@mdlayher Finally making the move to your `kobject` package.

File: internal/app/udevd/main.go [hunk 1]
```diff
@@ -10,50 +10,46 @@ import (
...
 func init() {
 	log.SetFlags(log.Lshortfile | log.Ldate | log.Lmicroseconds | log.Ltime)
 }
 
 func watch() error {
-	k, err := uevent.Dial()
+	client, err := kobject.New()
 	if err != nil {
-		return err
+		log.Fatal(err)
```
Comment (mdlayher) [inline internal/app/udevd/main.go]: Return error.

File: internal/app/udevd/main.go [hunk 2]
```diff
@@ -10,50 +10,46 @@ import (
...
-	defer k.Close()
-
-	for u := range k.Watch() {
-		if u.Error != nil {
-			log.Printf("uevent error: %+v\n", err)
-			continue
+	for {
+		event, err := client.Receive()
+		if err != nil {
+			panic(err)
```
Comment (mdlayher) [inline internal/app/udevd/main.go]: I'm not sure panic is what you really want in these two calls.

File: internal/app/udevd/main.go [hunk 3]
```diff
@@ -10,50 +10,46 @@ import (
...
 	}
-
-	return nil
 }
 
 // nolint: gocyclo
-func handle(message *uevent.Message) (err error) {
-	if message.Subsystem == uevent.SubsystemBlock {
+func handle(event *kobject.Event) (err error) {
+	switch e := event; {
```
Comment (mdlayher) [inline internal/app/udevd/main.go]: Nit: you can `switch e.Subsystem`.

File: internal/app/udevd/main.go (hunk 2 above)
Comment (andrewrynhard) [reply internal/app/udevd/main.go]: Ahh yes, I do not.

File: internal/app/udevd/main.go [hunk 4]
```diff
@@ -10,50 +10,45 @@ import (
...
-	defer k.Close()
-
-	for u := range k.Watch() {
-		if u.Error != nil {
-			log.Printf("uevent error: %+v\n", err)
-			continue
+	for {
+		event, err := client.Receive()
+		if err != nil {
+			log.Printf("failed to receive event: %v", err)
```
Comment (mdlayher) [inline internal/app/udevd/main.go]: You'll need a continue after this log.

File: internal/app/udevd/main.go [hunk 5]
```diff
@@ -10,50 +10,45 @@ import (
...
 	}
-
-	return nil
 }
 
 // nolint: gocyclo
-func handle(message *uevent.Message) (err error) {
-	if message.Subsystem == uevent.SubsystemBlock {
+func handle(event *kobject.Event) (err error) {
+	if event.Subsystem == "block" {
```
Comment (mdlayher) [inline internal/app/udevd/main.go]: Not: if you invert this if statement and do an early return, you can remove the indentation from this entire block.

File: internal/app/udevd/main.go (hunk 4 above)
Comment (andrewrynhard) [reply internal/app/udevd/main.go]: Crazy how sleep affects you when you don’t have it 😄. My newborn has been keeping me up all day every day. Thanks for catching.

Comment (mdlayher) [reply internal/app/udevd/main.go]: All good!

**Pull request merged**